lib/metadata/WebsiteConfiguration.js

```diff
@@ -0,0 +1,202 @@
+export class RoutingRule {
+    /**
+    * Represents a routing rule in a website configuration.
+    * @constructor
+    * @param {object} params - object containing redirect and condition objects
+    * @param {object} params.redirect - specifies how to redirect requests
+    * @param {string} [params.redirect.protocol] - protocol to use for redirect
+    * @param {string} [params.redirect.hostName] - hostname to use for redirect
+    * @param {string} [params.redirect.replaceKeyPrefixWith] - string to replace
+    *   keyPrefixEquals specified in condition
+    * @param {string} [params.redirect.replaceKeyWith] - string to replace key
+    * @param {string} [params.redirect.httpRedirectCode] - http redirect code
+    * @param {object} [params.condition] - specifies conditions for a redirect
+    * @param {string} [params.condition.keyPrefixEquals] - prefix in URI that
+    *   triggers a redirect
+    * @param {string} [params.condition.httpErrorCodeReturnedEquals] - http code
+    *   that triggers a redirect
+    */
+    constructor(params) {
+        if (params && params.redirect) {
+            this.setRedirect(params.redirect);
+        }
+        if (params && params.condition) {
+            this.setCondition(params.condition);
+        }
+    }
+
+    /**
+    * Set the condition object
+    * @param {object} obj - object to set as condition
+    * @return {undefined};
+    */
+    setCondition(obj) {
+        if (obj) {
+            const { keyPrefixEquals, httpErrorCodeReturnedEquals } = obj;
+            this.condition = { keyPrefixEquals, httpErrorCodeReturnedEquals };
+        } else {
```

Comment:
Turns out I did use the setters to create the RoutingRules before adding them to the WebsiteConfiguration instance while parsing, but I removed them.